### PR TITLE
Reduce logging of common redirects and expected items

### PR DESCRIPTION
**Pull request by ericholscher**
This should massively reduce the amount of logs we're showing:

Screenshot 2023-06-28 at 10 35 49 AM

Chart: https://chart-embed.service.newrelic.com/herald/026e34db-4e14-4867-8bce-85845402be24?height=400px&timepicker=true

### CODE REVIEW

File: readthedocs/proxito/views/serve.py [hunk 1]
```diff
@@ -691,7 +689,7 @@ def _get_custom_404_page(self, request, project, version=None):
                     storage_root_path, tryfile
                 )
                 if build_media_storage.exists(storage_filename_path):
-                    log.info(
+                    log.debug(
                         "Serving custom 404.html page.",
```
Comment (humitos) [inline readthedocs/proxito/views/serve.py:693]: Do we have too many of these? I don't think too many people have custom 404 pages. This is actually useful to know what those projects are.

Not super relevant, but just nothing that I've used this log line in the past.

**Review by humitos — APPROVED** [review]

File: readthedocs/proxito/views/serve.py (hunk 1 above)
Comment (ericholscher) [reply readthedocs/proxito/views/serve.py:693]: @humitos It's our most common log message, as shown in the chart above.

Comment (humitos) [reply readthedocs/proxito/views/serve.py:693]: Wow 😳. OK.

The chart didn't load the first time I opened the PR 🤷‍♂️

**Review by stsewd — APPROVED** [review]
There are some linting problems that need to be fixed

**Comment by ericholscher** [comment]
https://github.com/readthedocs/common/pull/180

**Comment by humitos** [comment]
I don't know where that "Log event" UI you showed in the description comes from, but I just wanted to comment here that I usually use the "Log Patterns" one: https://one.newrelic.com/logger/log-patterns?account=63767&duration=604800000&state=bfbf16ba-d4d1-b557-f420-a41585d0530b and has been being really useful to me.

**Comment by ericholscher** [comment]
@humitos nice, that's a good dashboard. Mine was just a custom query that I futzed with in NR -- I figured it was built in, but couldn't find it quickly :)

**Pull request merged**